### ISSUE TITLE
Blackpowder Fouling Extremely Rapid

### ISSUE THREAD

**Issue by Grubleafeater** [CleverRaven/Cataclysm-DDA#63515]
### Describe the bug

If placed in inventory, weapons that have been fired with EVEN ONCE with blackpowder will rust away in a matter of HOURS.

This is easiest to see with flintlock pistols or pistols you shoot with manually loaded black powder munitions, but happens to seemingly all guns.

### Attach save file

N/A

### Steps to reproduce

Spawn in pocket universe, 2 each of a handful of guns, and black powder ammo. Fire each once, drop 1 of the pair and put the other in pocket universe. Rest.

### Expected behavior

Having the weapon in inventory should not impact rust rate; also, _weapons should not rust to XX within 2 hours of being used once._

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.2364 (21H2)
- Game Version: 2ee4928 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Xedra Evolved [xedra_evolved],
    Magiclysm [magiclysm],
    Blaze Industries [blazeindustries],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    My Sweet Cataclysm [my_sweet_cataclysm]
]

**Comment by andrei8l** [comment]
/confirmed

Top-level items never process blackpowder fouling because the flag `BLACKPOWDER_FOULING_DAMAGE` doesn't mark them as active.

https://github.com/CleverRaven/Cataclysm-DDA/blob/43d20409b45fbb6b08cd7aaa39191e7f9d23a9db/src/item.cpp#L12522

Items stored in containers with at least one active item ignore that check and roll 1:2000 chance every turn for damage.

https://github.com/CleverRaven/Cataclysm-DDA/blob/43d20409b45fbb6b08cd7aaa39191e7f9d23a9db/src/item.cpp#L13389

#63233 will fix this bypass, but not for items in inventory.

So basically blackpowder fouling hasn't worked correctly at least since the pocket system was introduced.

**Comment by anoobindisguise** [comment]
Discussed this on the dev discord with OP and can confirm seeing the same effects in my game - also, should blackpowder fouling take an entire hour to clean? An hour is already extreme for a regular gun, but for blackpowder fouling you're not typically taking the whole gun apart and cleaning + lubricating each part, you're cleaning the corrosive element from the barrel.  Ideally there would be different clean times for a muzzleloader compared to a proper modern gun though.

**Comment by evankimori** [comment]
Did some more testing with this issue mentioned in my Issue #64067. See attached testing info in ticket. Not sure if my testing would be of any use here, please advise.
